src/Microsoft.AspNetCore.JsonPatch/JsonPatchDocumentOfT.cs

```diff
@@ -776,33 +772,24 @@ private string GetPropertyNameFromMemberExpression(MemberExpression memberExpres
             return null;
         }
 
-        private static bool ContinueWithSubPath(ExpressionType expressionType, bool firstTime)
+        private static bool ContinueWithSubPath(ExpressionType expressionType)
         {
-            if (firstTime)
-            {
-                return (expressionType == ExpressionType.ArrayIndex
-                       || expressionType == ExpressionType.Call
-                       || expressionType == ExpressionType.Convert
-                       || expressionType == ExpressionType.MemberAccess
-                       || expressionType == ExpressionType.Parameter);
-            }
-            else
-            {
-                return (expressionType == ExpressionType.ArrayIndex
-                    || expressionType == ExpressionType.Call
-                    || expressionType == ExpressionType.Convert
-                    || expressionType == ExpressionType.MemberAccess);
-            }
+            return (expressionType == ExpressionType.ArrayIndex
+                || expressionType == ExpressionType.Call
+                || expressionType == ExpressionType.Convert
+                || expressionType == ExpressionType.MemberAccess);
+
         }
 
-        private static string GetIndexerInvocation(Expression expression)
+        // Evaluates the value of the key or index which may be an int or a string, 
+        // or some other expression type.
+        // The expression is converted to executable code and the resulting delegate is returned as a string.
```

Comment:
`The expression is converted to a delegate and the result of executing the delegate is returned as a string`